pkg/agent/core/ngt/service/ngt.go

```diff
@@ -783,33 +768,57 @@ func (n *ngt) CreateAndSaveIndex(ctx context.Context, poolSize uint32) (err erro
 }
 
 func (n *ngt) Exists(uuid string) (oid uint32, ok bool) {
-	oid, ok = n.kvs.Get(uuid)
+	ok = n.vq.IVExists(uuid)
 	if !ok {
-		ok = n.vq.IVExists(uuid)
-	} else {
-		ok = !n.vq.DVExists(uuid)
+		oid, ok = n.kvs.Get(uuid)
+		if !ok {
+			log.Debugf("Exists\tuuid: %s's data not found in kvsdb and insert vqueue\terror: %v", uuid, errors.ErrObjectIDNotFound(uuid))
+			return 0, false
+		}
+		if n.vq.DVExists(uuid) {
+			log.Debugf("Exists\tuuid: %s's data found in kvsdb and not found in insert vqueue, but delete vqueue data exists. the object will be delete soon\terror: %v",
+				uuid, errors.ErrObjectIDNotFound(uuid))
+			return 0, false
+		}
 	}
 	return oid, ok
 }
 
+func (n *ngt) GetObject(uuid string) (vec []float32, err error) {
+	vec, ok := n.vq.GetVector(uuid)
+	if !ok {
+		oid, ok := n.kvs.Get(uuid)
+		if !ok {
+			log.Debugf("GetObject\tuuid: %s's data not found in kvsdb and insert vqueue", uuid)
+			return nil, errors.ErrObjectIDNotFound(uuid)
+		}
+		if n.vq.DVExists(uuid) {
+			log.Debugf("GetObject\tuuid: %s's data found in kvsdb and not found in insert vqueue, but delete vqueue data exists. the object will be delete soon", uuid)
+			return nil, errors.ErrObjectIDNotFound(uuid)
+		}
+		vec, err = n.core.GetVector(uint(oid))
+		if err != nil {
+			log.Debugf("GetObject\tuuid: %s oid: %d's vector not found in ngt index", uuid, oid)
+			return nil, errors.ErrObjectNotFound(err, uuid)
+		}
+	}
+	return vec, nil
+}
+
 func (n *ngt) readyForUpdate(uuid string, vec []float32) (err error) {
 	if len(uuid) == 0 {
 		return errors.ErrUUIDNotFound(0)
 	}
-	if len(vec) == 0 {
-		return errors.ErrInvalidDimensionSize(len(vec), 0)
+	if len(vec) != n.GetDimensionSize() {
+		return errors.ErrInvalidDimensionSize(len(vec), n.GetDimensionSize())
 	}
 	ovec, err := n.GetObject(uuid)
-	if err != nil || len(vec) != len(ovec) {
-		// if error (GetObject cannot find vector) or vector length is not equal let's try update
+	if err != nil ||
+		len(vec) != len(ovec) ||
```

Comment:
maybe this checking is not need? as it checked in line 812.
but having this checking is ok

Reply:
I think it's okay.